clang/lib/Headers/__clang_cuda_intrinsics.h

```diff
@@ -479,6 +479,275 @@ inline __device__ unsigned __funnelshift_rc(unsigned low32, unsigned high32,
   return ret;
 }
 
+#define INTRINSIC_LOAD(func_name, asm_op, decl_type, internal_type, asm_type) \
```

Comment:
We have to be careful with the names used in the headers we provide. Generally, all symbols we use, including macro arguments should be prefixed with `__`

We also often use `#pragma push_macro/pop_macro` to save/restore the preprocessor state once we're done handling the file. Search for it in the other headers in this directory.